manifests/artemis.pp

```diff
@@ -22,7 +22,7 @@
 
   file { "${candlepin::catalina_home}/conf/cert-users.properties":
     ensure  => file,
-    content => template('candlepin/tomcat/cert-users.properties.erb'),
+    content => Deferred('inline_epp', ['katelloUser=<%= $artemis_client_dn %>', {'artemis_client_dn' => $candlepin::artemis_client_dn}]),
```

Comment:
Does this work? Isn't `$candlepin::artemis_client_dn` already set or does it work because `$candlepin::artemis_client_dn` can be a `Deferred`?

Reply:
This works because it can be a Deferred, that is why I updated the acceptance test to ensure it works as my local testing showed that you can run into multiple weird situations with Deferred if you do not have them all in the right places.

Reply:
Ok, sounds good. The only note I'd have is that I don't think this inline epp template has a trailing newline. The current template does which can lead to a service restart even if it's not really needed. So I think this is needed.
```suggestion
    content => Deferred('inline_epp', ["katelloUser=<%= $artemis_client_dn %>\n", {'artemis_client_dn' => $candlepin::artemis_client_dn}]),
```